plugins/org.python.pydev.shared_ui/src/org/python/pydev/shared_ui/field_editors/JsonFieldEditor.java

```diff
@@ -0,0 +1,609 @@
+        this(name, labelText, UNLIMITED, parent);
+    }
+
+    @Override
+    protected void adjustForNumColumns(int numColumns) {
+        GridData gd = (GridData) textField.getLayoutData();
+        gd.horizontalSpan = numColumns - 1;
+        // We only grab excess space if we have to
+        // If another field editor has more columns then
+        // we assume it is setting the width.
+        gd.grabExcessHorizontalSpace = gd.horizontalSpan == 1;
+    }
+
+    /**
+     * Checks whether the text input field contains a valid value or not.
+     *
+     * @return <code>true</code> if the field value is valid,
+     *   and <code>false</code> if invalid
+     */
+    protected boolean checkState() {
+        setErrorMessage(JFaceResources.getString("StringFieldEditor.errorMessage"));
+
+        boolean result = false;
+
+        if (textField == null) {
+            setErrorMessage("JsonFieldEditor did not load properly.");
+            return handleErrorMessage(result);
+        } else {
+            String txt = textField.getText();
+            result = !txt.trim().isEmpty();
+            if (!result && !isEmptyStringAllowed()) {
+                return handleErrorMessage(result);
+            }
+        }
+
+        result = checkTextFieldJSON();
+        if (!result) {
+            return handleErrorMessage(result);
+        }
+
+        // call hook for subclasses
+        result = doCheckState();
+
+        return handleErrorMessage(result);
+    }
+
+    /**
+     * Checks whether the JSON text input field contains a valid value or not.
+     * 
+     * It also sets as red the entire line from <code>StyledText textField</code>
+     * where the JSON has a syntax error.
+     *
+     * @return <code>true</code> if the field value is valid,
+     *   and <code>false</code> if invalid
+     */
+    private boolean checkTextFieldJSON() {
+        PySelection sel = new PySelection(new Document(textField.getText()));
+        Tuple<Integer, Integer> errorPos = getJSONErrorPos(sel.getDoc().get());
+
+        if (errorPos == null) {
+            textField.setStyleRange(null);
+            return true;
+        }
+
+        int errorLine = errorPos.o1 - 1;
+        StyleRange styleRange = new StyleRange();
+        styleRange.start = sel.getLineOffset(errorLine);
+        styleRange.length = sel.getLine(errorLine).length();
+        styleRange.fontStyle = SWT.BOLD;
+        styleRange.foreground = this.textField.getDisplay().getSystemColor(SWT.COLOR_RED);
+        textField.setStyleRange(styleRange);
+
+        return false;
+    }
+
+    /**
+     * @param str string to validates JSON.
+     * @return returns <code>Tuple<Integer, Integer>(line,column)</code> 
+     * specifying that JSON has an error and it is in returned line and column,
+     *   and <code>null</code> if JSON is valid
+     */
+    private Tuple<Integer, Integer> getJSONErrorPos(String str) {
+        try {
+            JsonValue.readFrom(str);
```

Comment:
One note, I noted that you are doing additional validation by subclassing.

In general, I thing not having to use subclasses for that is nicer (so, you can configure it from the outside without subclassing).

One way to do that would be setting an `org.python.pydev.shared_core.callbacks.ICallback<Ret, Arg>` which receives the Json value and returns an error message or null.

Something as:

```
    public void setAdditionalJsonValidation(ICallback<Optional<String>, JsonValue> additionalValidation) {
        this.additionalValidation = additionalValidation;
    }

    // Example to set with lambda: 
    setAdditionalJsonValidation((json) -> {
        // Validate json and return eror
        if(...) {
            return Optional.of(errorMsg);
        }
        // No error
        return Optional.of(null);
    });
    
    // Example to use:
    Optional<String> optError = this.additionalValidation.call(jsonContents);
    if(optError.isPresent()) {
        // deal with error
        String errorMsg = optError.get();
    } else {
        // no error
    }

```